core/arch/arm/kernel/tee_ta_manager.c

```diff
@@ -626,57 +483,71 @@ static TEE_Result tee_ta_load(const kta_signed_header_t *signed_ta,
 	TAILQ_INIT(&ctx->cryp_states);
 	TAILQ_INIT(&ctx->objects);
 	TAILQ_INIT(&ctx->storage_enums);
-	ctx->head = sec_head;
 #if defined(CFG_SE_API)
 	ctx->se_service = NULL;
 #endif
 
-	/* by default NSec DDR: starts at TA function code. */
-	ctx->mem_swap = (uintptr_t)(nmem_ta + sizeof(ta_head_t) +
-			     sec_head->nbr_func * sizeof(ta_func_head_t));
+	res = tee_ta_load_elf(ctx, sec_signed_ta, signed_ta);
+	if (res != TEE_SUCCESS)
+		goto error_return;
+
+	ctx->load_addr = tee_mmu_get_load_addr(ctx);
+	ta_head = (struct ta_head *)(vaddr_t)ctx->load_addr;
 
-	ctx->num_res_funcs = ctx->head->zi_size >> 20;
-	ctx->head->zi_size &= 0xfffff;
-	if (ctx->num_res_funcs > ctx->head->nbr_func) {
-		res = TEE_ERROR_BAD_FORMAT;
+	if (memcmp(&ta_head->uuid, uuid, sizeof(TEE_UUID)) != 0) {
+		res = TEE_ERROR_SECURITY;
 		goto error_return;
 	}
 
-	res = tee_ta_load_user_ta(ctx, sec_head, nmem_ta, &heap_size);
-	if (res != TEE_SUCCESS)
+	/* check input flags bitmask consistency and save flags */
+	if ((ta_head->flags & opt_flags) != ta_head->flags ||
+	    (ta_head->flags & man_flags) != man_flags) {
+		EMSG("TA flag issue: flags=%x opt=%X man=%X",
+		     ta_head->flags, opt_flags, man_flags);
+		res = TEE_ERROR_BAD_FORMAT;
 		goto error_return;
+	}
+
+	ctx->flags = ta_head->flags;
+
+	/* Check if multi instance && single session config  */
+	if (((ctx->flags & TA_FLAG_SINGLE_INSTANCE) == 0) &&
+	    ((ctx->flags & TA_FLAG_MULTI_SESSION) == 0)) {
+		/*
+		 * assume MultiInstance/SingleSession,
+		 * same as MultiInstance/MultiSession
+		 */
```

Comment:
This is just copy paste from the old `tee_ta_load_user_ta()` function. If think we could remove this test all together as the multi-session property is ignored for for multi-instance TAs.